presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -170,10 +166,62 @@ public void setCursorName(String name)
     public boolean execute(String sql)
             throws SQLException
     {
+        clearCurrentResults();
+
         checkOpen();
-        // Only support returning a single result set
-        currentResult.set(executeQuery(sql));
-        return true;
+        checkNotNull(sql, "null sql statement");
+
+        StatementClient client = null;
+        ResultSet resultSet = null;
+        try {
+            client = connection().startQuery(sql);
+            resultSet = new PrestoResultSet(client);
+            QueryResults queryResults = client.current();
+
+            // query statement
+            if (queryResults.getUpdateType() == null) {
```

Comment:
check queryResults.getStats().getState() in case that query failed so you could throw some SQLException instead of returning empty result set